CRM/Extension/Mapper.php

```diff
@@ -301,28 +301,34 @@ public function getActiveModuleFiles($fresh = FALSE) {
         if (!empty($compat[$dao->full_name]['force-uninstall'])) {
           continue;
         }
-        try {
-          $moduleExtensions[] = [
-            'prefix' => $dao->file,
-            'filePath' => $this->keyToPath($dao->full_name),
-          ];
-        }
-        catch (CRM_Extension_Exception $e) {
-          // Putting a stub here provides more consistency
-          // in how getActiveModuleFiles when racing between
-          // dirty file-removals and cache-clears.
-          CRM_Core_Session::setStatus($e->getMessage(), '', 'error');
-          $moduleExtensions[] = [
-            'prefix' => $dao->file,
-            'filePath' => NULL,
-          ];
-        }
+        $moduleExtensions[] = [
+          'prefix' => $dao->file,
+          'fullName' => $dao->full_name,
+          'filePath' => NULL,
+        ];
       }
 
       if ($this->cache) {
         $this->cache->set($this->cacheKey . '_moduleFiles', $moduleExtensions);
       }
     }
+
+    // Since we're not caching the full path we add it now.
+    array_walk($moduleExtensions, function(&$value, $key) {
+      try {
+        if (!$value['filePath']) {
+          $value['filePath'] = $this->keyToPath($value['fullName']);
+        }
+      }
+      catch (CRM_Extension_Exception $e) {
+        // Putting a stub here provides more consistency
+        // in how getActiveModuleFiles when racing between
+        // dirty file-removals and cache-clears.
+        CRM_Core_Session::setStatus($e->getMessage(), '', 'error');
+        $value['filePath'] = NULL;
+      }
+    });
```

Comment:
This block is only needed once for each request? Otherwise, the failed extension will be caught multiple times.

Reply:
@agileware-pengyi I made `$moduleExtensions` a static variable which I am thinking will minimize the number of times this needs to be requested.

Reply:
@herbdool I see. But for the catch block, `filePath` is set to `NULL`, so it will be calculated next time when this function called.
It might not be a big issue, but the `CRM_Core_Session::setStatus` will be called multiple times for the same error.

Reply:
@agileware-pengyi the current code also set's `filePath` to NULL in a `while` loop so, as far as I can tell, my change isn't making it call it more times. Maybe fewer with the change to a static variable. If you still think it's a problem perhaps it can be addressed in a follow-up issue?

Reply:
@herbdool I might be wrong, but in the current code, the try&catch block only executes once (only if  `$moduleExtensions` is not cached).
Anyway, it is not an important issue. We can deal with it afterward.

Reply:
@agileware-pengyi I've made one more change based on @totten's suggestions. Now it will return early if the static variable is set so I think that makes more sense to me now. It will mean it'll only do the try/catch once per thread if I understand correctly.

Reply:
@herbdool thanks. I am good with it now